### PR TITLE
Real time data pt3

### PR DESCRIPTION
**Pull request by dasgarner**

### CODE REVIEW

**Comment by dasgarner** [comment]
TEST: /Schedule, /Campaign END

File: views/schedule-form-templates.twig [hunk 1]
```diff
@@ -135,8 +135,8 @@
         <div class="col-sm-2">
             <label for="criteria_type[]">
                 <select class="form-control" name="criteria_type[]">
-                    <option value="Custom" {{#eq type "Custom"}}selected{{/eq}}>{% endverbatim %}{% trans "Custom" %}{% verbatim %}</option>
-                    <option value="Weather" {{#eq type "Weather"}}selected{{/eq}}>{% endverbatim %}{% trans "Weather" %}{% verbatim %}</option>
+                    <option value="custom" {{#eq type "Custom"}}selected{{/eq}}>{% endverbatim %}{% trans "Custom" %}{% verbatim %}</option>
```
Comment (PeterMis) [inline views/schedule-form-templates.twig:138]: Should we change the handlebars condition to be lowercase `custom` and `weather` in this case? 🤔

**Review by PeterMis — CHANGES_REQUESTED** [review]
Looks good to me, just one small comment below

File: lib/Entity/Schedule.php [hunk 2]
```diff
@@ -841,11 +865,23 @@ public function save($options = [])
...
             $criteria->eventId = $this->eventId;
             $criteria->save();
+
+            $criteriaIds[] = $criteria->id;
         }
 
+        // Remove records that no longer exist.
+        $this->getStore()->update('
+            DELETE FROM `schedule_criteria` 
+             WHERE `id` NOT IN (' . implode(',', $criteriaIds) . ')
```
Comment (PeterMis) [inline lib/Entity/Schedule.php:879]: This probably wants to be executed if we do have any `$criteriaIds` to not upset sql

**Comment by dasgarner** [comment]
TEST: /Schedule, /Campaign END

**Review by PeterMis — APPROVED** [review]

**Pull request merged**